src/svg-renderer.js

```diff
@@ -390,11 +391,24 @@ class SvgRenderer {
             this._drawFromImage(scale, onFinish);
         } else {
             const img = new Image();
-            img.onload = () => {
+            const svgText = this.toString(true /* shouldInjectFonts */);
+            const doDraw = () => {
                 this._cachedImage = img;
                 this._drawFromImage(scale, onFinish);
             };
-            const svgText = this.toString(true /* shouldInjectFonts */);
+            img.onload = () => {
+                // In Safari, if the svg contains an image tag,
+                // the image tag may not be rendered when the onload event fires.
+                if (isSafari) {
+                    const div = document.createElement('div');
+                    div.innerHTML = svgText;
+                    Promise.all([...div.querySelectorAll('image')].map($img => new Promise(resolve => {
+                        const innerImage = document.createElement('img');
+                        innerImage.onerror = innerImage.onload = resolve;
+                        innerImage.src = $img.getAttribute('xlink:href');
```

Comment:
Due to https://github.com/LLK/scratch-assets/issues/63, Scratch won't be supporting SVGs with the xlink:href attribute, so unfortunately I think we can't take this fix.

Reply:
That's a shame.😢

Reply:
LLK / scratch-assets＃63 can't be open @fsih

Reply:
@fsih How do you plan to handle converting bitmap sprites to vector? *Every* SVG `<image>` element requires `xlink:href`, including every bitmap sprite converted to vector.

Reply:
Maybe you're thinking of `href`s which point to external resources, but embedding a base64-encoded image requires `xlink:href` too.

Reply:
@Ryuurock Ah sorry, it's a private repo. The bug is that xlink:href that points to a URL can be misused as a security vulnerability to get the user's browser to request a URL.
@adroitwhiz That's a great point! The current fix would break bitmaps in vector assets. I will bring it up on the PR.

Reply:
@fsih What is our next action?

Reply:
@Ryuurock if you could file an issue on this repo with the gif you made that would be great: https://github.com/LLK/scratch-svg-renderer/issues We can investigate if there's a good way to fix this issue there.